File: 2017/code/strings/strings.py
#!/usr/bin/python3
""" Python strings """

def main():
    """ Main func """
    hello = "Hello"
    world = "World"

    print(hello)
    print(world)

    # retazce sa daju spajat
    print(hello + world)

    # aj ulozit do premenny
    concat = hello + world
    print(concat)

    #alebo pridat medzera
    print(hello + " " + world)

    #alebo inak
    print(" ".join([hello, world]))

    helloworld = hello + " " + world

    print(helloworld)

    # retazce sa daju slicovat
    print(helloworld[:5])
    print(helloworld[5:])

    # alebo indexovat
    print(hello[4] + world[0] + hello[2])

    # mozu sa opakovat
    print("*" * 80)
    print("*" + " " * 78 + "*")
    print(("*" + " " * 78 + "*\n") * 2, end='')
    print("*" + " "* int(78 / 2 - len(helloworld) / 2) + helloworld + " " * int(78 / 2 - len(helloworld) / 2 + 1) + "*")
    print(("*" + " " * 78 + "*\n") * 3, end='')
    print("*" * 80)

    # vedia kreslit
    base = 30
    for row in range(1, 30, 2):
        print(" " * int(base / 2 - row / 2) + "+" * row)

    #da sa v nich vyhladavat
    print("Je hello v Hello World?")
    print("hello" in helloworld)

    # pozor vyhladavanie je case-sensitive
    # tak este raz
    print("Je hello v Hello World?")
    print("hello" in helloworld.lower())

if __name__ == '__main__':
    main()

File: 2017/code/strings/strings.java
import java.lang.System;

import java.lang.String;

import static java.util.stream.Collectors.*;
import java.util.stream.Stream;


class StringSample
{
    public static String repeat(String str, int times)
    {
        // potrebuje java 8 a vyssiu
        return Stream.generate( () -> str ).limit( times ).collect( joining() );
    }

    public static void main(String[] args) {
        String hello = "Hello";
        String world = "World";

        System.out.println( hello );
        System.out.println( world );

        // retazce sa daju lahko spajat
        System.out.println( hello + world );

        // aj ulozit do novej premmenej
        String concat = hello + world;
        System.out.println( concat );

        // alebo pridat medzera
        System.out.println( hello + " " + world );

        // alebo aj inak od java 8
        //String.join(" ", Collections.nCopies(n, s));

        String helloworld = hello + " " + world;

        // retazce sa daju slicovat
        System.out.println( helloworld.substring( 0, 5) );
        System.out.println( helloworld.substring( 5 ) );

        // retazce sa daju indexovat
        System.out.println( hello.charAt( 4 ) + world.charAt( 0 ) + hello.charAt( 2 ) );

        // mozu sa opakovat ale treba to nakodovat
        System.out.println( repeat( "*", 80 ) );
        System.out.print( repeat( "*" + repeat(" ", 78) + "*\n", 2) );
        System.out.println( "*" + repeat( " ", 78 / 2 - helloworld.length() / 2 - 1 ) + helloworld + repeat( " ", 78 / 2 - helloworld.length() / 2 ) + "*" );
        System.out.print( repeat( "*" + repeat(" ", 78) + "*\n", 3) );
        System.out.println( repeat( "*", 80 ) );

        // vedia kreslit
        int base = 30;
        for( int row = 1; row <= base; row += 2 )
        {
            System.out.println( repeat( " ", base / 2 - row / 2 ) + repeat( "*", row ) );
        }

        //da sa v nich vyhladavat
        System.out.println( "Je hello v Hello World?" );
        System.out.println( helloworld.contains( "hello" ) );

        // pozor vyhladavanie je case-sensitive
        // tak este raz
        System.out.println( "Je hello v Hello World?" );
        System.out.println( helloworld.toLowerCase().contains( "hello" ) );
    }
} 
